app/decorators/flight_decorator.rb

```diff
@@ -0,0 +1,35 @@
+class FlightDecorator
+  attr_reader :component
+
+  def initialize(component)
+    @component = component
+  end
+
+  def departure_d
+    d_date = DateTime.parse(component.departure_datetime.to_s)
+    d_date.strftime("%A, #{d_date.day.ordinalize} %b %Y")
+  end
+
+  def departure_t
+    d_date = DateTime.parse(component.departure_datetime.to_s)
+    d_date.strftime("%l:%M %P")
+  end
+
+  def arrival_d
+    d_date = Date.parse(component.arrival_datetime.to_s)
+    d_date.strftime("%A, #{d_date.day.ordinalize} %b %Y")
+  end
+
+  def arrival_t
+    d_date = Time.parse(component.arrival_datetime.to_s)
```

Comment:
Do not use `Time.parse` without zone. Use one of `Time.zone.parse`, `Time.parse.current`, `Time.parse.in_time_zone`, `Time.parse.utc`, `Time.parse.getlocal`, `Time.parse.iso8601`, `Time.parse.jisx0301`, `Time.parse.rfc3339`, `Time.parse.to_i`, `Time.parse.to_f` instead.